core/shared/src/main/scala/fs2/async/Ref.scala

```diff
@@ -129,11 +156,17 @@ final class Ref[F[_],A] private[fs2] (implicit F: Effect[F], ec: ExecutionContex
       case Some(changeAndB) => F.pure(changeAndB)
     }
 
+  // TODO might want to eliminate this, since one can just fork at call site
+  // however is part of SyncOps since SyncRef can implement it more efficiently
+  // so if the two Refs stay separate this should stay.
   override def setAsyncPure(a: A): F[Unit] =
-    F.delay { actor ! Msg.Set(a, () => ()) }
+    async.fork {
+      F.async[Unit] { cb => set(a, () => cb(Right(()))) }
+    }
 
+  // TODO not sure why the shift here
   override def setSyncPure(a: A): F[Unit] =
-    F.async[Unit](cb => actor ! Msg.Set(a, () => cb(Right(())))) *> F.shift
+    F.async[Unit](cb => set(a, () => cb(Right(())))) *> F.shift
```

Comment:
I think I just added this shift when doing the `SyncRef` PR. Without it a test was hanging -- I think because the subsequence actions flatmapped on to the returned `F[Unit]` were getting run on the actor thread (based on where the callback was being invoked). Perhaps this isn't necessary if we are consistent in only ever invoking callbacks from within an EC.